nvtabular/ops/categorify.py

```diff
@@ -775,15 +836,41 @@ def _encode(
         if list_col:
             codes = cudf.DataFrame({selection_l[0]: gdf[selection_l[0]].list.leaves})
             codes["order"] = cp.arange(len(codes))
+            if buckets and selection_l[0] in buckets:
+                na_sentinel = _hash_bucket(gdf, buckets, selection_l[0])
         else:
-            codes = cudf.DataFrame({"order": cp.arange(len(gdf))})
+            codes = cudf.DataFrame({"order": cp.arange(len(gdf))}, index=gdf.index)
             for c in selection_l:
                 codes[c] = gdf[c].copy()
-        labels = codes.merge(
-            value, left_on=selection_l, right_on=selection_r, how="left"
-        ).sort_values("order")["labels"]
-        labels.fillna(na_sentinel, inplace=True)
-        labels = labels.values
+                if buckets and c in buckets and encode_type == "joint":
+                    na_sentinel = _hash_bucket(gdf, buckets, c)
+                elif buckets and encode_type == "combo":
+                    if len(selection_l) > 1:
+                        na_sentinel = _hash_bucket(
+                            gdf, buckets, tuple(selection_l), encode_type="combo"
+                        )
+                    else:
+                        na_sentinel = _hash_bucket(gdf, buckets, selection_l, encode_type="combo")
+        # apply frequency hashing
+        if freq_threshold and buckets and storage_name in buckets:
+            merged_df = codes.merge(
+                value, left_on=selection_l, right_on=selection_r, how="left"
+            ).sort_values("order")
+            merged_df.reset_index(drop=True, inplace=True)
```

Comment:
@benfred this is necessary because index of merged df from line 856-858 sort of messed, thus, yields wrong results after doing `merged_df["labels"].fillna(cudf.Series(na_sentinel + max_id + 1), inplace=True)`. Therefore, resetting index for merged_df before solves that issue.